pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -182,6 +187,17 @@ func (e *HookExecutor) executeExecNewPod(hook *deployapi.LifecycleHook, deployme
 waitLoop:
 	for {
 		updatedPod = nextPod()
+		if len(updatedPod.Status.ContainerStatuses) > 0 {
+			restarts := updatedPod.Status.ContainerStatuses[0].RestartCount
+			if hook.ExecNewPod.MaxRestarts > 0 && restarts >= hook.ExecNewPod.MaxRestarts {
+				glog.V(0).Infof("Maximum (%d) lifecycle pod %s/%s restarts reached, deleting pod", restarts, pod.Namespace, pod.Name)
+				updatedPod.Spec.RestartPolicy = kapi.RestartPolicyNever
```

Comment:
Why are you setting this here?  You're never using the `updatedPod` since you're returning from this block, right?